stdlib/hashtbl.ml

```diff
@@ -46,16 +46,8 @@ let flip_ongoing_traversal h =
 
 (* To pick random seeds if requested *)
 
-let randomized_default =
-  let params =
-    try Sys.getenv "OCAMLRUNPARAM" with Not_found ->
-    try Sys.getenv "CAMLRUNPARAM" with Not_found -> "" in
-  String.contains params 'R'
-
-let randomized = Atomic.make randomized_default
-
-let randomize () = Atomic.set randomized true
-let is_randomized () = Atomic.get randomized
+external randomize : unit -> unit = "caml_hashtbl_randomize"
+external is_randomized : unit -> bool = "caml_hashtbl_is_randomized"
```

Comment:
I'm nervous about replacing the atomic with an entire primitive call (even if we were to mark `caml_hashtbl_is_randomized` as `noalloc`). I think in the ideal world, we'd have just `caml_params->init_hashtbl_randomized` and then `caml_hashtbl_get_init_randomized` returning that and then we'd update `caml_runtime_parameters` to take `bool` instead of a `unit` ... i.e. we'd still have the atomic here, but `Sys.runtime_parameters` would instead be:

```ocaml
external runtime_parameters : bool -> string = "caml_runtime_parameters"
let runtime_parameters () = runtime_parameters (Hashtbl.is_randomised ())
```

but that would introduce a cycle. So I think instead - with plenty of comment - we can have _both_ the atomic here and the atomic in C... and so have something like:

```ocaml
let randomize () =
  Atomic.set randomized true;
  (* There is technically a race here where Hashtbl.create () creates randomized hash tables,
   but Sys.runtime_parameters doesn't yet return R=1. We choose not to care - Hashtbl.is_randomized will
   always return the correct value, and making Sys.runtime_parameters always be in sync would either
   add a C call to every Hashtbl.create call or would introduce a complicated dependency cycle between Sys
   and Hashtbl *)
  caml_hashtbl_randomize ();
```

I'm not 100% convinced...